File: client/src/app/entities/branchrole.ts
import {User} from './user';
import {Branch} from './branch';
import {Department} from './department';
import {DataPage} from '../shared/data-page';
import {Branchrolestatus} from './branchrolestatus';

export class Branchrole {
  id: number;
  code: string;
  branch: Branch;
  name: string;
  min: number;
  max: number;
  branchrolestatus: Branchrolestatus;
  allowance: number;
  department: Department;
  creator: User;
  tocreation: string;
}

export class BranchroleDataPage extends DataPage{
    content: Branchrole[];
}

File: server/src/main/java/bit/project/server/entity/Branchrole.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import javax.persistence.*;
import javax.persistence.Id;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Branchrole{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private String name;

    private Integer min;

    private Integer max;

    private BigDecimal allowance;

    private LocalDateTime tocreation;


    @ManyToOne
    private Branch branch;

    @ManyToOne
    private Branchrolestatus branchrolestatus;

    @ManyToOne
    private Department department;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    @JsonIgnore
    @OneToMany(mappedBy = "branchrole")
    private List<Branchroleassignment> branchroleBranchroleassignmentList;


    public Branchrole(Integer id){
        this.id = id;
    }

    public Branchrole(Integer id, String code, String name, Department department){
        this.id = id;
        this.code = code;
        this.name = name;
        this.department = department;
    }

}